Kconfig.zephyr

```diff
@@ -738,6 +738,24 @@ endmenu
 
 menu "Boot Options"
 
+config APPLICATION_VERSION_SUPPORTED
+	bool
+	help
+	  Hidden option used to indicate that the current image supports
+	  version information which may be embedded into the application or
+	  used for signing.
+
+config APPLICATION_VERSION
+	string "Application version"
+	default "0.0.0+0"
+	depends on APPLICATION_VERSION_SUPPORTED
+	help
+	  This specifies the version of the application, when signing images
+	  using imgtool, this is the version that will be used.
```

Comment:
The Kconfig names are too generic for something so tool-specific IMO. Why not something like IMGTOOL_APP_VERSION?

Reply:
The idea is to use this for other systems too e.g. for the mcuboot version itself and, for ncs, the nsib application version - so is not imgtool specific (just the default version string is)